src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -0,0 +1,632 @@
+    //   chainPromise.push(getWalletBalance(chain.chainId, chain.title))
+    // });
+    return await Promise.allSettled(chainPromise).catch((e) => {
+      return [];
+    });
+  }
+
+  const fetchAllAccountBalances = async () => {
+    var accountBalanceWithSupportedChains = await getBalanceForAllChains().then((data) => data.map((d) => d.value));
+    accountBalanceWithSupportedChains = accountBalanceWithSupportedChains?.filter(
+      (data:any) => data.keyBasedWallet > 0 || data.smartWallet,
+    );
+  
+    const totalOfKeyBased = accountBalanceWithSupportedChains.reduce((accumulator, object) => {
+      return accumulator + object.keyBasedUSD;
+    }, 0);
+    const totalOfSmartWallet = accountBalanceWithSupportedChains.reduce((accumulator, object) => {
+      return accumulator + object.smartWalletUSD;
+    }, 0);
+  
+    const totalKeyBasedPLRTokens = accountBalanceWithSupportedChains?.reduce((accumulator, object) => {
+      return accumulator + object.keyBasedWallet;
+    }, 0);
+
+    const totalSmartWalletPLRTokens = accountBalanceWithSupportedChains?.reduce((accumulator, object) => {
+      return accumulator + object.smartWallet;
+    }, 0);
+
+    setKeyBasedTotal(totalOfKeyBased);
+    setSmartWalletTotal(totalOfSmartWallet);
+    setTotalKeyBasedPLRTokens(totalKeyBasedPLRTokens);
+    setTotalSmartWalletPLRTokens(totalSmartWalletPLRTokens);
+    setAccounts(accountBalanceWithSupportedChains);
+  };
+
+  useEffect(() => {
+    fetchAllAccountBalances();
+  }, []);
+
+  const updateAvailableOffers = useCallback<
+    () => Promise<ExchangeOffer[] | undefined>
+  >(
+    debounce(async () => {
+      // there is a race condition here
+      if (multiCallData && fixed) {
+        return;
+      }
+      setSelectedOffer(null);
+      setAvailableOffers([]);
+
+      if (
+        !sdk ||
+        !selectedFromAsset ||
+        !amount ||
+        !selectedFromNetwork?.chainId ||
+        !isValidAmount(amount)
+      )
+        return;
+
+      setIsLoadingAvailableOffers(true);
+
+      try {
+        // needed computed account address before calling getExchangeOffers
+        if (!accountAddress) await sdk.computeContractAccount();
+        const offers = await sdk.getExchangeOffers({
+          fromChainId: selectedFromAsset.chainId,
+          fromAmount: ethers.utils.parseUnits(
+            amount,
+            selectedFromAsset.decimals
+          ),
+          toTokenAddress: hasEnoughPLR ? plrDaoMemberNFT.address : plrDaoAsset.address,
+          fromTokenAddress: selectedFromAsset.address,
+        });
+        return offers;
+      } catch (e) {
+        //
+      }
+    }, 200),
+    [sdk, selectedFromAsset, amount, selectedFromNetwork, accountAddress, selectedAccountType]
+  );
+
+  useEffect(() => {
+    accountAddress || providerAddress && sdk?.getNftList({
+      account: accountAddress || providerAddress,
+    }).then((output:NftList) => {
+      if(output?.items?.length) {
```

Comment:
This endpoint can be error prone like the others, it's worth checking you have a valid response here before determining items length etc